src/Sarif.Converters/AndroidStudioConverter.cs

```diff
@@ -150,51 +152,73 @@ public Result ConvertProblemToSarifResult(AndroidStudioProblem problem)
+                //}
             }
 
             result.Locations = new List<Location> { location };
 
             return result;
         }
 
-        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem)
+        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem, out int index)
         {
-            string parentLogicalLocationKey = null;
+            index = -1;
+            string fullyQualifiedName = null;
+            string delimiter = string.Empty;
+
+            if (!string.IsNullOrEmpty(problem.Module))
+            {
+                index = AddLogicalLocation(index, ref fullyQualifiedName, problem.Module, LogicalLocationKind.Module, delimiter);
+                delimiter = @"\";
+            }
 
-            parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.Module, LogicalLocationKind.Module);
-            parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.Package, LogicalLocationKind.Package);
+            if (!string.IsNullOrEmpty(problem.Package))
+            {
+                index = AddLogicalLocation(index, ref fullyQualifiedName, problem.Package, LogicalLocationKind.Package, delimiter);
+                delimiter = @"\";
+            }
 
             if (problem.EntryPointName != null)
             {
                 if ("class".Equals(problem.EntryPointType, StringComparison.OrdinalIgnoreCase))
                 {
-                    parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.EntryPointName, LogicalLocationKind.Type);
+                    index = AddLogicalLocation(index, ref fullyQualifiedName, problem.EntryPointName, LogicalLocationKind.Type, delimiter);
+
                 }
                 else if ("method".Equals(problem.EntryPointType, StringComparison.OrdinalIgnoreCase))
                 {
-                    parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.EntryPointName, LogicalLocationKind.Member);
+                    index = AddLogicalLocation(index, ref fullyQualifiedName, problem.EntryPointName, LogicalLocationKind.Member, delimiter);
                 }
+                // TODO: 'file' is another entry point type. should we be doing something here?
             }
 
-            return parentLogicalLocationKey;
+            return fullyQualifiedName;
         }
 
-        private string AddLogicalLocation(string parentKey, string value, string kind, string delimiter = @"\")
+        private int AddLogicalLocation(int parentIndex, ref string fullyQualifiedName, string value, string kind, string delimiter = ".")
         {
-            if (!String.IsNullOrEmpty(value))
+            fullyQualifiedName = fullyQualifiedName + delimiter + value;
+
+            // Need to decide which item gets preference, name vs. fully qualified name
+            // if both match. The behaviors of various API in the SDK differs on this point.
```

Comment:
>match [](start = 23, length = 5)

As noted elsewhere: the spec says that if `name` and `fullyQualifiedName` match (that is, if this is a top-level logical location), the `name` is required (as it _always_ is), and `fullyQualifiedLogicalName` is optional.

Reply:
tracking item created as noted

---
In reply to: [243866740](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243866740) [](ancestors = 243866740)